File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/PostConditionFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.ConversionProcess;
import de.unikassel.vs.alica.planDesigner.alicamodel.PostCondition;
import org.w3c.dom.Element;

public class PostConditionFactory extends Factory {

    public static PostCondition create(Element postConditionNode, ConversionProcess cp) {
        PostCondition postCondition = new PostCondition();
        ConditionFactory.fillConditon(postConditionNode, postCondition, cp);
        return postCondition;
    }
}

File: alica_engine/src/engine/modelmanagement/factories/PostConditionFactory.cpp
#include "engine/modelmanagement/factories/PostConditionFactory.h"
#include "engine/model/PostCondition.h"
#include "engine/modelmanagement/Strings.h"
#include "engine/modelmanagement/factories/ConditionFactory.h"

namespace alica
{
PostCondition* PostConditionFactory::create(const YAML::Node& postConditionNode, AbstractPlan* plan)
{
    PostCondition* postCondition = new PostCondition();
    ConditionFactory::fillCondition(postConditionNode, postCondition, plan);
    return postCondition;
}
} // namespace alica
